torch/distributions/distribution.py

```diff
@@ -60,3 +65,37 @@ def enumerate_support(self):
             Variable or Tensor iterating over dimension 0.
         """
         raise NotImplementedError
+
+    def _extended_shape(self, sample_shape=()):
+        """
+        Returns the size of the sample returned by the distribution, given
+        a `sample_shape`. Note, that the batch and event shapes of a distribution
+        instance are fixed at the time of construction. If this is empty, the
+        returned shape is upcast to (1,).
+
+        Args:
+            sample_shape (torch.Size): the size of the sample to be drawn.
+        """
+        shape = sample_shape + self._batch_shape + self._event_shape
+        if not shape:
+            shape = (1,)
+        return shape
+
+    def _validate_log_prob_arg(self, value):
+        """
+        Argument validation for `log_prob` methods. The rightmost dimensions
+        of a value to be scored via `log_prob` must agree with the distribution's
+        batch and event shapes.
+
+        Args:
+            value (Tensor or Variable): the tensor whose log probability is to be
+                computed by the `log_prob` method.
+        Raises
+            ValueError: when the rightmost dimensions of `value` do not match the
+                distribution's batch and event shapes.
+        """
+        if not isinstance(value, (torch.Tensor, Variable)):
```

Comment:
I think you want `if not (torch.is_tensor(value) or isinstance(value, Variable))`, since `torch.Tensor` is an alias for a single type, rather than a base class.

Reply:
I guess that's fine - if we have a tuple in the second argument to `isinstance`, the function returns True if value is a subclass of any of the classes in the tuple. e.g.
```
> isinstance(2, (str, int))
True
```

Reply:
```py
>>> isinstance(torch.DoubleTensor([0]), torch.Tensor)
False
>>> isinstance(torch.FloatTensor([0]), torch.Tensor)
True
```

Reply:
Ahh..I see. I hadn't realized this. Let me go and change all the places where we have this.